pkg/operator/render.go

```diff
@@ -156,3 +160,79 @@ func isSingleStackIPv6(serviceCIDRs []string) (bool, error) {
 func (rc renderConfig) GenerateProxyCookieSecret() string {
 	return base64.StdEncoding.EncodeToString([]byte(utilrand.String(32)))
 }
+
+func onPremPlatformShortName(cfg mcfgv1.ControllerConfigSpec) interface{} {
+	if cfg.Infra.Status.PlatformStatus != nil {
+		switch cfg.Infra.Status.PlatformStatus.Type {
+		case configv1.BareMetalPlatformType:
+			return "kni"
+		case configv1.OvirtPlatformType:
+			return "ovirt"
+		case configv1.OpenStackPlatformType:
+			return "openstack"
+		case configv1.VSpherePlatformType:
+			return "vsphere"
+		default:
+			return ""
+		}
+	} else {
+		return ""
+	}
+}
+
+func onPremPlatformIngressIP(cfg mcfgv1.ControllerConfigSpec) (interface{}, error) {
+	if cfg.Infra.Status.PlatformStatus != nil {
+		switch cfg.Infra.Status.PlatformStatus.Type {
+		case configv1.BareMetalPlatformType:
+			return cfg.Infra.Status.PlatformStatus.BareMetal.IngressIP, nil
+		case configv1.OvirtPlatformType:
+			return cfg.Infra.Status.PlatformStatus.Ovirt.IngressIP, nil
+		case configv1.OpenStackPlatformType:
+			return cfg.Infra.Status.PlatformStatus.OpenStack.IngressIP, nil
+		case configv1.VSpherePlatformType:
+			return cfg.Infra.Status.PlatformStatus.VSphere.IngressIP, nil
+		default:
+			return nil, fmt.Errorf("invalid platform for API VIP")
+		}
+	} else {
+		return nil, fmt.Errorf("")
+	}
+}
+
+func onPremPlatformNodeDNSIP(cfg mcfgv1.ControllerConfigSpec) (interface{}, error) {
```

Comment:
Is this used? I'm not sure we need it now that we removed the DNS VIP.

Reply:
Yeah. I haven't seen it in any manifests/templates. I'll remove it.